controller/internal/routes/posture_check_api_model.go

```diff
@@ -103,5 +138,63 @@ func MapPostureCheckToRestEntity(_ *env.AppEnv, _ *response.RequestContext, e mo
 }
 
 func MapPostureCheckToRestModel(i *model.PostureCheck) (rest_model.PostureCheckDetail, error) {
-	return nil, nil
+	var ret rest_model.PostureCheckDetail
+
+	switch subType := i.SubType.(type) {
+	case *model.PostureCheckOperatingSystem:
+		osArray := rest_model.OperatingSystemArray{}
+
+		for _, osMatch := range subType.OperatingSystems {
+			osArray = append(osArray, &rest_model.OperatingSystem{
+				Type:     rest_model.OsType(osMatch.OsType),
+				Versions: osMatch.OsVersions,
+			})
+		}
+
+		ret = &rest_model.PostureCheckOperatingSystemDetail{
+			OperatingSystems: osArray,
+		}
+
+		setBaseEntityDetailsOnPostureCheck(ret, i)
+
+	case *model.PostureCheckProcess:
+		processMatch := &rest_model.Process{
+			Hashes:            subType.Hashes,
+			OsType:            rest_model.OsType(subType.OperatingSystem),
+			Path:              &subType.Path,
+			SignerFingerprint: subType.Fingerprint,
+		}
+
+		ret = &rest_model.PostureCheckProcessDetail{
+			Process: processMatch,
+		}
+
+		setBaseEntityDetailsOnPostureCheck(ret, i)
+	case *model.PostureCheckWindowsDomains:
+		ret = &rest_model.PostureCheckDomainDetail{
+			Domains: subType.Domains,
+		}
+		setBaseEntityDetailsOnPostureCheck(ret, i)
+	case *model.PostureCheckMacAddresses:
+		ret = &rest_model.PostureCheckMacAddressDetail{
+			MacAddresses: subType.MacAddresses,
+		}
+		setBaseEntityDetailsOnPostureCheck(ret, i)
+	}
+
+	return ret, nil
+}
+
+func setBaseEntityDetailsOnPostureCheck(check rest_model.PostureCheckDetail, i *model.PostureCheck) {
+	createdAt := strfmt.DateTime(i.CreatedAt)
```

Comment:
Glorious hacky workaround for a go-swagger bug.

Reply:
might i request this glorious comment be placed into the source code for posterity?

Reply:
It is in the swagger file.